src/buttons/AddPhotosphere.tsx

```diff
@@ -193,6 +165,68 @@ function AddPhotosphere({
           Cancel
         </Button>
       </Stack>
+
+      <Dialog
+        open={mapDialogOpen}
+        onClose={() => {
+          setMapDialogOpen(false);
+        }}
+      >
```

Comment:
This dialog and the `handleMapClick` function are repeated here and in `App.tsx`.
It should be abstracted into one component that can be referenced in both locations.